windows/quickstart-bazel build fails storage quickstart with v1.24.0 release

build passes using v1.23.0 release.

[log from failed build]
2021-02-02T04:03:52.2342302+00:00  Fetch dependencies for storage [1]

Starting local Bazel server and connecting to it...

INFO: Invocation ID: b8cb5743-99d4-427e-8894-c619e1493d57

Loading: 0 packages loaded

Loading: 0 packages loaded

Loading: 0 packages loaded

Loading: 0 packages loaded

Loading: 0 packages loaded

Loading: 0 packages loaded

Loading: 0 packages loaded

Loading: 0 packages loaded

Loading: 4 packages loaded

Loading: 4 packages loaded

Loading: 5 packages loaded

Loading: 5 packages loaded

Loading: 18 packages loaded

DEBUG: C:/b/bwoopu2y/external/bazel_tools/tools/cpp/lib_cc_configure.bzl:118:10: 

Auto-Configuration Warning: neither 'TMP' nor 'TEMP' environment variables are set, using 'C:\Windows\Temp' as default

Loading: 38 packages loaded

INFO: All external dependencies fetched successfully.

Loading: 44 packages loaded

Loading: 44 packages loaded



2021-02-02T04:04:57.1076160+00:00  Compiling quickstart for storage

INFO: Invocation ID: 769fec44-3f85-4e5a-8c46-b08eaa08cb58

Loading: 

Loading: 0 packages loaded

Analyzing: target //:quickstart (0 packages loaded, 0 targets configured)

DEBUG: C:/b/bwoopu2y/external/bazel_tools/tools/cpp/lib_cc_configure.bzl:118:10: 

Auto-Configuration Warning: neither 'TMP' nor 'TEMP' environment variables are set, using 'C:\Windows\Temp' as default

WARNING: T:/src/github/google-cloud-cpp/google/cloud/storage/quickstart/BUILD:19:10: in cc_binary rule //:quickstart: target '//:quickstart' depends on deprecated target '@com_github_googleapis_google_cloud_cpp//:storage_client': this target will be removed on or around 2022-02-15, please use //:storage instead.

INFO: Analyzed target //:quickstart (2 packages loaded, 1468 targets configured).



INFO: Found 1 target...

[0 / 40] [Prepa] BazelWorkspaceStatusAction stable-status.txt

[3 / 41] checking cached actions

[4 / 41] checking cached actions

[5 / 42] checking cached actions

[16 / 59] Compiling quickstart.cc; 0s local, remote-cache ... (5 actions running)

[28 / 79] Compiling quickstart.cc; 2s local, remote-cache ... (5 actions, 4 running)

[41 / 85] Compiling quickstart.cc; 3s local, remote-cache ... (5 actions, 4 running)

[58 / 105] Compiling quickstart.cc; 4s local, remote-cache ... (5 actions, 4 running)

[76 / 151] Compiling external/com_google_absl/absl/strings/internal/str_format/float_conversion.cc; 0s remote-cache ... (4 actions running)

[101 / 151] Compiling external/boringssl/src/ssl/ssl_cipher.cc; 0s remote-cache ... (4 actions running)

[134 / 155] Linking external/com_google_absl/absl/strings/strings.lib; 0s remote-cache ... (4 actions, 3 running)

[163 / 240] Compiling external/com_github_googleapis_google_cloud_cpp/google/cloud/storage/internal/curl_handle.cc; 1s local, remote-cache ... (6 actions, 5 running)

ERROR: C:/b/bwoopu2y/external/com_github_googleapis_google_cloud_cpp/google/cloud/storage/BUILD:67:11: C++ compilation of rule '@com_github_googleapis_google_cloud_cpp//google/cloud/storage:google_cloud_cpp_storage' failed (Exit 1)

C:\b\bwoopu2y\execroot\com_github_googleapis_google_cloud_cpp_storage_quickstart\external\com_github_googleapis_google_cloud_cpp\google\cloud\storage\oauth2\google_application_default_credentials_file.cc : fatal error C1083: Cannot open compiler generated file: '': Invalid argument

Target //:quickstart failed to build

Use --verbose_failures to see the command lines of failed build steps.

INFO: Elapsed time: 20.820s, Critical Path: 7.35s

INFO: 139 processes: 138 remote cache hit, 1 local.

FAILED: Build did NOT complete successfully

FAILED: Build did NOT complete successfully

bazel test failed with exit code 1.



2021-02-02T04:05:18.2153220+00:00 Disk(s) size and space for troubleshooting

DeviceID    : C:

DriveType   : 3

VolumeName  : 

FreeSpaceGB : 223.95

Capacity    : 300.00



DeviceID    : T:

DriveType   : 3

VolumeName  : tmpfs

FreeSpaceGB : 99.62

Capacity    : 99.98



Build failed with exit code 1

--- a/google/cloud/bigtable/quickstart/BUILD
+++ b/google/cloud/bigtable/quickstart/BUILD
@@ -22,6 +22,6 @@ cc_binary(
         "quickstart.cc",
     ],
     deps = [
-        "@com_github_googleapis_google_cloud_cpp//:bigtable_client",
+        "@com_github_googleapis_google_cloud_cpp//:bigtable",
     ],
 )

--- a/google/cloud/bigtable/quickstart/WORKSPACE
+++ b/google/cloud/bigtable/quickstart/WORKSPACE
@@ -14,7 +14,7 @@
 
 # A minimal WORKSPACE file showing how to use the Google Cloud Bigtable C++
 # client library in Bazel-based projects.
-workspace(name = "com_github_googleapis_google_cloud_cpp_bigtable_quickstart")
+workspace(name = "google_cloud_cpp_bigtable_quickstart")
 
 # Add the necessary Starlark functions to fetch google-cloud-cpp.
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

--- a/google/cloud/pubsub/quickstart/BUILD
+++ b/google/cloud/pubsub/quickstart/BUILD
@@ -22,6 +22,6 @@ cc_binary(
         "quickstart.cc",
     ],
     deps = [
-        "@com_github_googleapis_google_cloud_cpp//:pubsub_client",
+        "@com_github_googleapis_google_cloud_cpp//:pubsub",
     ],
 )

--- a/google/cloud/pubsub/quickstart/WORKSPACE
+++ b/google/cloud/pubsub/quickstart/WORKSPACE
@@ -14,7 +14,7 @@
 
 # A minimal WORKSPACE file showing how to use the Google Cloud Pub/Sub C++
 # client library in Bazel-based projects.
-workspace(name = "com_github_googleapis_google_cloud_cpp_pubsub_quickstart")
+workspace(name = "google_cloud_cpp_pubsub_quickstart")
 
 # Add the necessary Starlark functions to fetch google-cloud-cpp.
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

--- a/google/cloud/spanner/quickstart/BUILD
+++ b/google/cloud/spanner/quickstart/BUILD
@@ -20,6 +20,6 @@ cc_binary(
         "quickstart.cc",
     ],
     deps = [
-        "@com_github_googleapis_google_cloud_cpp//:spanner_client",
+        "@com_github_googleapis_google_cloud_cpp//:spanner",
     ],
 )

--- a/google/cloud/spanner/quickstart/WORKSPACE
+++ b/google/cloud/spanner/quickstart/WORKSPACE
@@ -14,7 +14,7 @@
 
 # A minimal WORKSPACE file showing how to use the Google Cloud Spanner C++
 # client library in Bazel-based projects.
-workspace(name = "com_github_googleapis_google_cloud_cpp_spanner_quickstart")
+workspace(name = "google_cloud_cpp_spanner_quickstart")
 
 # Add the necessary Starlark functions to fetch google-cloud-cpp.
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

--- a/google/cloud/storage/quickstart/BUILD
+++ b/google/cloud/storage/quickstart/BUILD
@@ -22,6 +22,6 @@ cc_binary(
         "quickstart.cc",
     ],
     deps = [
-        "@com_github_googleapis_google_cloud_cpp//:storage_client",
+        "@com_github_googleapis_google_cloud_cpp//:storage",
     ],
 )

--- a/google/cloud/storage/quickstart/WORKSPACE
+++ b/google/cloud/storage/quickstart/WORKSPACE
@@ -14,7 +14,7 @@
 
 # A minimal WORKSPACE file showing how to use the Google Cloud Storage C++
 # client library in Bazel-based projects.
-workspace(name = "com_github_googleapis_google_cloud_cpp_storage_quickstart")
+workspace(name = "google_cloud_cpp_storage_quickstart")
 
 # Add the necessary Starlark functions to fetch google-cloud-cpp.
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
@@ -23,9 +23,9 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 # NOTE: Update this version and SHA256 as needed.
 http_archive(
     name = "com_github_googleapis_google_cloud_cpp",
-    sha256 = "914c9596ee9f271a4ba2de701388009d1f6a7eb0ea269d625aae06be1a51ee9e",
-    strip_prefix = "google-cloud-cpp-1.23.0",
-    url = "https://github.com/googleapis/google-cloud-cpp/archive/v1.23.0.tar.gz",
+    sha256 = "705992bbf5d86a5d5b4276fe249ca495bc0827f1835cb433f3f6be777072aa62",
+    strip_prefix = "google-cloud-cpp-1.24.0",
+    url = "https://github.com/googleapis/google-cloud-cpp/archive/v1.24.0.tar.gz",
 )
 
 # Load indirect dependencies due to

